woocommerce/woocommerce
# Update WCPay banners for WooPay

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Update WCPay promotions on WooPay eligible countries to show WooPay references.

This change is related to 2020-gh-Automattic/woopay



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Follow 17784-gh-Automattic/woocommerce.com testing instructions.
2. Update [this line](https://github.com/woocommerce/woocommerce/blob/5efff2c8a28556f7d61e058cb483d8616421a5a9/plugins/woocommerce/src/Internal/Admin/WCPayPromotion/WCPayPromotionDataSourcePoller.php#L17) to `https://woocommerce.test/wp-json/wccom/payment-gateway-suggestions/1.0/payment-method/promotions.json`, or `http://host.docker.internal/wp-json/wccom/payment-gateway-suggestions/1.0/payment-method/promotions.json` if you are using Docker.
3. Comment out [this line](https://github.com/woocommerce/woocommerce/blob/9fec692865cbdcf61f011ab5d292555f694c28a8/plugins/woocommerce/src/Admin/DataSourcePoller.php#L111) and [this line](https://github.com/woocommerce/woocommerce/blob/9fec692865cbdcf61f011ab5d292555f694c28a8/plugins/woocommerce/src/Admin/DataSourcePoller.php#L115) to ignore cache.
4. Set your WooCommerce store country to US.
5. With WCPay uninstalled (you can just rename the plugin folder) and all payment methods disabled, access `WooCommerce > Overview > Set up payments`, it should show this banner with WooPay references

<img width="718" alt="Screenshot 2023-08-07 at 3 03 03 PM" src="https://github.com/woocommerce/woocommerce/assets/1693223/1c2f924f-e7dc-46b7-b4c3-9a8ed2c4831b">

6. Access `WooCommerce > Settings > Payments` page, it should show a description mentioning WooPay.

<img width="984" alt="Screenshot 2023-08-07 at 1 57 27 PM" src="https://github.com/woocommerce/woocommerce/assets/1693223/e804c55d-b5e3-4892-9c4f-37069fc702e6">

7. Change the store country to a country that supports WCPay but not WooPay (ex. Canada).
8. Reload the `Set up payments` page, it should not mention WooPay.

<img width="718" alt="Screenshot 2023-08-07 at 3 02 22 PM" src="https://github.com/woocommerce/woocommerce/assets/1693223/8b3c7c4e-6272-4cb6-9e7c-d224709d4e76">

9. Reload the Payments settings page, it should not mention WooPay.

<img width="1028" alt="Screenshot 2023-08-07 at 2 57 51 PM" src="https://github.com/woocommerce/woocommerce/assets/1693223/ef1b6b2e-311e-423e-9f57-daf4ed932685">

10. Install WCPay but deactivate it.
11. Change your store country back to US.
12. Access `WooCommerce > Overview`.
13. The `Set up payments` section should reference WooPay merchant terms.

<img width="758" alt="Screenshot 2023-08-07 at 2 55 04 PM" src="https://github.com/woocommerce/woocommerce/assets/1693223/0bf550fd-d5d1-4149-960b-c0ba4fa51042">

14. Change the store country to a country that supports WCPay but not WooPay (ex. Canada).
15. Reload the `WooCommerce > Overview` page.
16. It should not reference WooPay merchant terms.

<img width="727" alt="Screenshot 2023-08-07 at 2 58 06 PM" src="https://github.com/woocommerce/woocommerce/assets/1693223/d5e1fc87-9e2b-4de4-82a4-d00b1125c2a3">

17. Change the country to US.
17. Activate WCPay but disable it as a payment gateway on the `WooCommerce > Settings > Payments` page.
18. Reload the page.
19. The promotion banner should mention WooPay merchant terms.

<img width="1093" alt="Screenshot 2023-08-07 at 3 56 56 PM" src="https://github.com/woocommerce/woocommerce/assets/1693223/d328655f-4b60-4865-9428-4898791726cb">

21. Change the store country to a country that supports WCPay but not WooPay (ex. Canada).
22. Reload the `WooCommerce > Settings > Payments` page.
23. The promotion banner should not mention WooPay merchant terms.

<img width="1275" alt="Screenshot 2023-08-07 at 3 46 18 PM" src="https://github.com/woocommerce/woocommerce/assets/1693223/67376259-55b1-40b7-977d-2d0bebcb293d">

24. WooPay should never be mentioned on old WooCommerce versions even if it's on a eligible country, you can change the WC version [here](https://github.com/woocommerce/woocommerce/blob/5efff2c8a28556f7d61e058cb483d8616421a5a9/plugins/woocommerce/woocommerce.php#L6) to an older version and repeat the eligible steps mentioned here and it should act as it was ineligible.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
